README.md
# Legal Document Deliveries - Operations plugin for WordPress

[![Codeship status](https://www.codeship.io/projects/TBD/status)](https://www.codeship.io/projects/TBD)
[![ProjectStatus](http://stillmaintained.com/michael-cannon/ldd-operations.png)](http://stillmaintained.com/michael-cannon/ldd-operations)

## Welcome to the Legal Document Deliveries - Operations Github repository

Legal Document Deliveries - Operations plugin allows you to add custom fields to the edit screen bulk and quick edit panels.

While the documentation for the [Legal Document Deliveries - Operations plugin](http://wordpress.org/plugins/ldd-operations/) can be found at WordPress.org Plugin Directory, here you can browse the source of the project, find and discuss open issues and, even [contribute yourself](https://github.com/michael-cannon/ldd-operations/blob/master/CONTRIBUTING.md).

## Installation

Here's a [guide on how to install Legal Document Deliveries - Operations in your WordPress site](http://wordpress.org/plugins/ldd-operations/installation/). If you want to run the Git version though, you have two options:

* You can clone the GitHub repository: https://github.com/michael-cannon/ldd-operations.git
* Download it directly as a ZIP file: https://github.com/michael-cannon/ldd-operations/archive/master.zip

This will download the latest development version of Legal Document Deliveries - Operations by Michael Cannon. While this version is usually stable, it is not recommended for use in a production environment.

## Bugs

If you find an issue, [let us know here](https://github.com/michael-cannon/ldd-operations/issues/new)!

## Support

This is a developer's portal for Legal Document Deliveries - Operations by Michael Cannon and should not be used for support. Please visit the [Legal Document Deliveries - Operations Knowledge Base](https://aihrus.zendesk.com/categories/20102742) for frequently asked questions, offering ideas, or getting support.

## Contributions

Anyone is welcome to contribute to Legal Document Deliveries - Operations. Please [read the guidelines](https://github.com/michael-cannon/ldd-operations/blob/master/CONTRIBUTING.md) for contributing to this repository.

There are various ways you can contribute:

* [Raise an issue](https://github.com/michael-cannon/ldd-operations/issues) on GitHub.
* Fork Legal Document Deliveries - Operations and send us a Pull Request with your bug fixes and/or new features.
* [Translate Legal Document Deliveries - Operations by Michael Cannon into different languages](https://aihrus.zendesk.com/entries/23691557).
* Provide feedback and [suggestions on enhancements](https://github.com/michael-cannon/ldd-operations/issues?direction=desc&labels=Enhancement&page=1&sort=created&state=open).

## Thank You
* Thank you [WordPress SEO by Yoast](https://github.com/jdevalk/wordpress-seo/blob/master/README.md) for this README.md and like inspirations.


CHANGELOG.md
# CHANGELOG Legal Document Deliveries - Operations

## master

## 1.0.0
* Abstract mailer operations for easier reuse
* Add agent meta box
* Add delivery details to delivery record
* Add ldd_delivery_progress template marker
* Add notice_new_delivery, notice_assign_agent, notice_status_change methods
* Add notify_cc
* Add page count and delivery ids
* Add point of contact details to delivery record
* Add status updates to delivery records as comments
* Add time elasped helpers
* Capture pre save current_delivery_data
* Consolidate email body content
* Disable client notification during agent assigned
* Disable donate and premium links
* Enable global actions
* Initial code release 
* Notify client of delivery
* PDF counter for small ones
* Relates docs to delivery record
* Remove admin only restriction
* Remove mass processor
* Remove Process plugin action link
* Rename some template markers
* RESOLVE Duplicate mailer content
* Send new_delivery_notice
* Set plugin URL to aihr.us

tests/tests/bootstrap.php
<?php

require_once getenv( 'WP_TESTS_DIR' ) . '/includes/functions.php';

function _manually_load_plugin() {
	require dirname( __FILE__ ) . '/../../ldd-operations.php';
}


tests_add_filter( 'muplugins_loaded', '_manually_load_plugin' );

require getenv( 'WP_TESTS_DIR' ) . '/includes/bootstrap.php';
?>


prepare-plugin.sh
#!/bin/bash

NEW_ABBR="LDD_OPERATIONS_"
NEW_BASE="ldd-operations"
NEW_CLASS="LDD_Operations"
NEW_FILTER="${NEW_ABBR,,}"
NEW_KB_PATH=""
NEW_SITE="https://github.com/michael-cannon/${NEW_BASE}"
NEW_SITE=
NEW_SLUG="${NEW_FILTER}"
NEW_SLUG_LONG="${NEW_BASE/-/_}"
NEW_TITLE="Legal Document Deliveries - Operations"
NEW_TITLE_SHORT="LDD Operations"

OLD_ABBR="LDD_ORDERING_"
OLD_BASE="ldd-ordering"
OLD_CLASS="LDD_Ordering"
OLD_FILTER="${OLD_ABBR,,}"
OLD_KB_PATH=""
OLD_SITE="https://github.com/michael-cannon/${OLD_BASE}"
OLD_SLUG="${OLD_FILTER}"
OLD_SLUG_LONG="${OLD_BASE/-/_}"
OLD_TITLE="Legal Document Deliveries - Ordering"
OLD_TITLE_SHORT="LDD Ordering"

echo
echo "Begin converting ${OLD_TITLE} to ${NEW_TITLE} plugin"

FILES=`find . -type f \( -name "*.css" -o -name "*.md" -o -name "*.php" -o -name "*.txt" -o -name "*.xml" \)`
for FILE in ${FILES} 
do
	if [[ '' != ${NEW_ABBR} ]]
	then
		perl -pi -e "s#${OLD_ABBR}#${NEW_ABBR}#g" ${FILE}
		perl -pi -e "s#${NEW_ABBR}_#${NEW_ABBR}#g" ${FILE}
	fi

	if [[ '' != ${NEW_BASE} ]]
	then
		perl -pi -e "s#${OLD_BASE}#${NEW_BASE}#g" ${FILE}
	fi

	if [[ '' != ${NEW_CLASS} ]]
	then
		perl -pi -e "s#${OLD_CLASS}#${NEW_CLASS}#g" ${FILE}
	fi

	if [[ '' != ${NEW_FILTER} ]]
	then
		perl -pi -e "s#${OLD_FILTER}#${NEW_FILTER}#g" ${FILE}
	fi

	if [[ '' != ${NEW_KB_PATH} ]]
	then
		perl -pi -e "s#${OLD_KB_PATH}#${NEW_KB_PATH}#g" ${FILE}
	fi

	if [[ '' != ${NEW_SITE} ]]
	then
		perl -pi -e "s#${OLD_SITE}#${NEW_SITE}#g" ${FILE}
	fi

	if [[ '' != ${NEW_SLUG} ]]
	then
		perl -pi -e "s#${OLD_SLUG}#${NEW_SLUG}#g" ${FILE}
		perl -pi -e "s#${NEW_SLUG}_#${NEW_SLUG}#g" ${FILE}
	fi

	if [[ '' != ${NEW_SLUG_LONG} ]]
	then
		perl -pi -e "s#${OLD_SLUG_LONG}#${NEW_SLUG_LONG}#g" ${FILE}
	fi

	if [[ '' != ${NEW_TITLE} ]]
	then
		perl -pi -e "s#${OLD_TITLE}#${NEW_TITLE}#g" ${FILE}
	fi

	if [[ '' != ${NEW_TITLE_SHORT} ]]
	then
		perl -pi -e "s#${OLD_TITLE_SHORT}#${NEW_TITLE_SHORT}#g" ${FILE}
	fi
done

if [[ -e 000-code-qa.txt ]]
then
	rm 000-code-qa.txt
fi

mv ${OLD_BASE}.php ${NEW_BASE}.php
mv assets/css/${OLD_BASE}.css assets/css/${NEW_BASE}.css
mv includes/class-${OLD_BASE}-settings.php includes/class-${NEW_BASE}-settings.php
mv includes/class-${OLD_BASE}-widget.php includes/class-${NEW_BASE}-widget.php
mv includes/class-${OLD_BASE}.php includes/class-${NEW_BASE}.php
mv languages/${OLD_BASE}.pot languages/${NEW_BASE}.pot

if [[ -e .git ]]
then
	rm -rf .git
fi

LIB_AIHRUS="includes/libraries/aihrus"
if [[ -e ${LIB_AIHRUS} ]]
then
	rm -rf ${LIB_AIHRUS}
fi

git init
git add *
git add .gitignore
git commit -m "Initial plugin creation"
git remote add origin git@github.com:michael-cannon/${NEW_BASE}.git
echo "git push origin master"

includes/functions.php
<?php
/**
 * Copyright 2014 Michael Cannon (email: mc@aihr.us)
 * This program is free software; you can redistribute it and/or modify
 * it under the terms of the GNU General Public License, version 2, as
 * published by the Free Software Foundation.
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
 * GNU General Public License for more details.
 * You should have received a copy of the GNU General Public License
 * along with this program; if not, write to the Free Software
 * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA
 */

if ( ! function_exists( 'ldd_operations_time_elasped' ) ) {
	function ldd_operations_time_elasped( $args, $field, $meta ) {
		$delivery_id = get_the_ID();

		echo LDD_Operations::time_elasped( $delivery_id );
	}
}


if ( ! function_exists( 'ldd_operations_time_elasped_current' ) ) {
	function ldd_operations_time_elasped_current( $args, $field, $meta ) {
		$delivery_id = get_the_ID();

		echo LDD_Operations::time_elasped( $delivery_id, false );
	}
}

?>


includes/requirements.php
<?php
/*
	Copyright 2014 Michael Cannon (email: mc@aihr.us)

	This program is free software; you can redistribute it and/or modify
	it under the terms of the GNU General Public License, version 2, as
	published by the Free Software Foundation.

	This program is distributed in the hope that it will be useful,
	but WITHOUT ANY WARRANTY; without even the implied warranty of
	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
	GNU General Public License for more details.

	You should have received a copy of the GNU General Public License
	along with this program; if not, write to the Free Software
	Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
 */


function ldd_operations_requirements_check() {
	$valid_requirements = true;
	if ( ! is_plugin_active( LDD_OPERATIONS_REQ_BASE ) ) {
		$valid_requirements = false;
		add_action( 'admin_notices', 'ldd_operations_notice_version' );
	}

	if ( ! $valid_requirements ) {
		deactivate_plugins( LDD_OPERATIONS_BASE );
	}

	return $valid_requirements;
}


function ldd_operations_notice_version() {
	aihr_notice_version( LDD_OPERATIONS_REQ_BASE, LDD_OPERATIONS_REQ_NAME, LDD_OPERATIONS_REQ_SLUG, LDD_OPERATIONS_REQ_VERSION, LDD_OPERATIONS_NAME );
}

?>


ldd-operations.php
<?php
/**
 * Plugin Name: Legal Document Deliveries - Operations
 * Plugin URI: https://github.com/michael-cannon/ldd-operations
 * Description: LDD operations system helper
 * Version: 1.0.0
 * Author: Michael Cannon
 * Author URI: http://aihr.us/resume/
 * License: GPLv2 or later
 * Text Domain: ldd-operations
 * Domain Path: /languages
 */


/**
 * Copyright 2014 Michael Cannon (email: mc@aihr.us)
 * This program is free software; you can redistribute it and/or modify
 * it under the terms of the GNU General Public License, version 2, as
 * published by the Free Software Foundation.
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
 * GNU General Public License for more details.
 * You should have received a copy of the GNU General Public License
 * along with this program; if not, write to the Free Software
 * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA
 */

if ( ! defined( 'LDD_OPERATIONS_BASE' ) )
	define( 'LDD_OPERATIONS_BASE', plugin_basename( __FILE__ ) );

if ( ! defined( 'LDD_OPERATIONS_DIR' ) )
	define( 'LDD_OPERATIONS_DIR', plugin_dir_path( __FILE__ ) );

if ( ! defined( 'LDD_OPERATIONS_DIR_INC' ) )
	define( 'LDD_OPERATIONS_DIR_INC', LDD_OPERATIONS_DIR . 'includes/' );

if ( ! defined( 'LDD_OPERATIONS_DIR_LIB' ) )
	define( 'LDD_OPERATIONS_DIR_LIB', LDD_OPERATIONS_DIR_INC . 'libraries/' );

if ( ! defined( 'LDD_OPERATIONS_NAME' ) )
	define( 'LDD_OPERATIONS_NAME', 'Legal Document Deliveries - Operations' );

if ( ! defined( 'LDD_OPERATIONS_REQ_BASE' ) )
	define( 'LDD_OPERATIONS_REQ_BASE', 'ldd/ldd.php' );

if ( ! defined( 'LDD_OPERATIONS_REQ_NAME' ) )
	define( 'LDD_OPERATIONS_REQ_NAME', 'Legal Document Deliveries - Core ' );

if ( ! defined( 'LDD_OPERATIONS_REQ_SLUG' ) )
	define( 'LDD_OPERATIONS_REQ_SLUG', 'ldd' );

if ( ! defined( 'LDD_OPERATIONS_REQ_VERSION' ) )
	define( 'LDD_OPERATIONS_REQ_VERSION', '1.0.0' );

if ( ! defined( 'LDD_OPERATIONS_VERSION' ) )
	define( 'LDD_OPERATIONS_VERSION', '1.0.0' );

require_once LDD_OPERATIONS_DIR_INC . 'requirements.php';

global $ldd_operations_activated;

$ldd_operations_activated = true;
if ( ! ldd_operations_requirements_check() ) {
	$ldd_operations_activated = false;

	return false;
}

require_once LDD_OPERATIONS_DIR_INC . 'functions.php';
require_once LDD_OPERATIONS_DIR_INC . 'class-ldd-operations.php';


add_action( 'plugins_loaded', 'ldd_operations_init', 99 );


/**
 *
 *
 * @SuppressWarnings(PHPMD.LongVariable)
 * @SuppressWarnings(PHPMD.UnusedLocalVariable)
 */
if ( ! function_exists( 'ldd_operations_init' ) ) {
	function ldd_operations_init() {
		if ( LDD_Operations::version_check() ) {
			global $LDD_Operations;
			if ( is_null( $LDD_Operations ) )
				$LDD_Operations = new LDD_Operations();
			
			do_action( 'ldd_operations_init' );
		}
	}
}


register_activation_hook( __FILE__, array( 'LDD_Operations', 'activation' ) );
register_deactivation_hook( __FILE__, array( 'LDD_Operations', 'deactivation' ) );
register_uninstall_hook( __FILE__, array( 'LDD_Operations', 'uninstall' ) );


if ( ! function_exists( 'ldd_operations_shortcode' ) ) {
	function ldd_operations_shortcode( $atts ) {
		return LDD_Operations::ldd_operations_shortcode( $atts );
	}
}

?>
